installation_and_upgrade/ibex_install_utils/install_tasks.py

```diff
@@ -920,6 +917,81 @@ def create_journal_sql_schema(self):
                                                    os.getenv("MYSQL_PASSWORD", "environment variable not set"))
                 RunProcess(SYSTEM_SETUP_PATH, "add_journal_table.bat", prog_args=[sql_password]).run()
 
+    @contextmanager
+    def timestamped_pv_backups_file(self, name, directory, extension="txt"):
+        """
+        Context manager to create a timestamped file in the pv backups directory
+
+        Args:
+            name (str): path to the file
+            extension (str): the extension of the file
+        """
+        filename = os.path.join(PV_BACKUPS_DIR, directory, "{}_{}.{}"
+                                .format(name, datetime.today().strftime('%Y-%m-%d-%H-%M-%S'), extension))
+
+        if not os.path.exists(os.path.join(PV_BACKUPS_DIR, directory)):
+            os.makedirs(os.path.join(PV_BACKUPS_DIR, directory))
+
+        with open(filename, "w") as f:
+            yield f
+
+    def save_motor_parameters_to_file(self):
+        """
+        Saves the motor parameters to csv file.
+        """
+        with Task("Save motor parameters to csv file", self._prompt) as task:
+            if task.do_step:
+                with self.timestamped_pv_backups_file(name="motors", directory="motors", extension="csv") as f:
+                    get_params_and_save_to_file(f)
+
+    def save_blocks_to_file(self):
+        """
+        Saves block parameters in a file.
+        """
+
+        with Task("Save block parameters to file", self._prompt) as task:
+            if task.do_step:
+                blocks = self._ca.get_blocks()
+
+                if blocks is None:
```

Comment:
g.get_blocks() returns an empty list if there are no blocks.
In python an empty list is not equivalent to None.
So here when there are no blocks, "if blocks is None:" will go to the else statement rather than telling the user that no blocks exist.

Reply:
Hi, this is a condition that occurs if the PV is disconnected (and therefore the value comes back as `None`) rather than the PV being connected but empty.

I've edited the documentation in CaWrapper to make it more obvious that it can return `None` on failure to connect to PV.